src/esi/Module.cc

```diff
@@ -24,34 +22,29 @@ static ESIParser::Register *prExpat = 0;
 
 void Esi::Init()
 {
-    assert(!prCustom); // we should be called once
-
-    prCustom = new ESIParser::Register("custom", &ESICustomParser::NewParser);
-
 #if HAVE_LIBXML2
+    assert(!prLibxml); // we should be called once
     prLibxml = new ESIParser::Register("libxml2", &ESILibxml2Parser::NewParser);
 #endif
 
 #if HAVE_LIBEXPAT
+    assert(!prLibxml); // we should be called once
     prExpat = new ESIParser::Register("expat", &ESIExpatParser::NewParser);
 #endif
 }
 
 void Esi::Clean()
 {
-    assert(prCustom); // we should be called once, and only after Init()
-
 #if HAVE_LIBEXPAT
+    assert(prExpat); // we should be called once, and only after Init()
```

Comment:
I disagree with the old comment. Or, to be more precise, I disagree that "should" is assert().

There is no reason to restrict cleanup that much IMHO. Let's not add these two asserts in Clean().

Reply:
done